torch/utils/tensorboard/summary.py

```diff
@@ -306,10 +306,9 @@ def make_video(tensor, fps):
 
     # encode sequence of images into gif string
     clip = mpy.ImageSequenceClip(list(tensor), fps=fps)
-    with tempfile.NamedTemporaryFile() as f:
-        filename = f.name + '.gif'
 
-    try:
+    filename = tempfile.NamedTemporaryFile(suffix='.gif').name
```

Comment:
Definitely clearer - thank you. However, it will still go away immediately, so let's change this to

```
filename = tempfile.NamedTemporaryFile(suffix='.gif', delete=False).name
```
Looks like you have all the removal logic below.

Reply:
Also, let's remove the `try ... except` around the `os.remove(filename)` below. Let's make sure it exists.